tests/misc_tests/test_pretrained_models.py

```diff
@@ -0,0 +1,564 @@
+import functools
+import unittest
+
+import chainer
+import chainer.functions as F
+from chainer import links as L
+from chainer import optimizers
+from chainer import testing
+import numpy as np
+
+import chainerrl
+from chainerrl import agents
+from chainerrl import explorers
+from chainerrl import links
+from chainerrl import policies
+from chainerrl.optimizers import rmsprop_async
+from chainerrl.action_value import DiscreteActionValue
+from chainerrl.q_functions import DistributionalDuelingDQN
+from chainerrl.misc import download_model
+from chainerrl import replay_buffer
+from chainerrl import v_functions
+
+
+@testing.parameterize(*testing.product(
+    {
+        'pretrained_type': ["best", "final"],
+    }
+))
+class TestLoadDQN(unittest.TestCase):
+
+    def _test_load_dqn(self, gpu):
+        q_func = links.Sequence(
+            links.NatureDQNHead(),
+            L.Linear(512, 4),
+            DiscreteActionValue)
+
+        opt = optimizers.RMSpropGraves(
+            lr=2.5e-4, alpha=0.95, momentum=0.0, eps=1e-2)
+        opt.setup(q_func)
+
+        rbuf = replay_buffer.ReplayBuffer(100)
+
+        explorer = explorers.LinearDecayEpsilonGreedy(
+            start_epsilon=1.0, end_epsilon=0.1,
+            decay_steps=10 ** 6,
+            random_action_func=lambda: np.random.randint(4))
+
+        agent = agents.DQN(q_func, opt, rbuf, gpu=gpu, gamma=0.99,
+                           explorer=explorer, replay_start_size=50,
+                           target_update_interval=10 ** 4,
+                           clip_delta=True,
+                           update_interval=4,
+                           batch_accumulator='sum',
+                           phi=lambda x: x)
+
+        model, exists = download_model("DQN", "BreakoutNoFrameskip-v4",
+                                       model_type=self.pretrained_type)
+        agent.load(model)
+        assert exists
```

Comment:
Why existence of cache is asserted? It should be false when running this test for the first time.

Reply:
The reason I assert for the existence of the cache is to ensure that we use the cached models during the PFN CI. When running this test locally, indeed it should be false. But when running the test on the CI, we pretrained models should already be downloaded. I omit this test from the Travis CI and only run it in the PFN CI.

Reply:
Tests should be able to run locally. Can you remove the assertions?

Reply:
Makes sense. I'll remove the assertions. But then we won't have the jobs fail if they caching of trained models isn't done properly, so we should just be vigilant and pay attention to ensure that we're not accidentally downloading models instead of using cached ones.

Reply:
That should be ensured definitely, but without breaking the tests locally. I think it is good to have a env var like CHAINERRL_ASSERT_DOWNLOADED_MODEL_IS_CACHED so that we can enable the assertion optionally.

Reply:
Done.